pkg/cache/ring.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2022 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import (
+	"bytes"
+
+	"github.com/tikv/pd/pkg/btree"
+)
+
+// Ring is a buffer, the key range must be continuous.
```

Comment:
Why call ring? I misunderstand it as a ring buffer.

Reply:
In my opinion, the ring describe all the items are continuous, not allow gaps.

Reply:
I think it's not a suitable name, `ring buffer` is a fixed-size buffer.

Reply:
yes, I will rename it as bucket-tree.